src/applauncher/000-vertical.qml

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (C) 2025 Timo Könnecke <github.com/eLtMosen>
+ *               2025 Darrel Griët <dgriet@gmail.com>
+ *               2015 Florent Revest <revestflo@gmail.com>
+ *               2014 Aleksi Suomalainen <suomalainen.aleksi@gmail.com>
+ *               2012 Timur Kristóf <venemo@fedoraproject.org>
+ *               2011 Tom Swindell <t.swindell@rubyx.co.uk>
+ * All rights reserved.
+ *
+ * BSD License (same as provided)
+ */
+
+import QtQuick 2.9
+import QtGraphicalEffects 1.12
+import org.asteroid.controls 1.0
+import Nemo.DBus 2.0
+
+Item {
+    id: root
+    property alias currentIndex: appsListView.currentIndex
+    property alias count: appsListView.count
+    property int lastLaunchedIndex: 0
+    property int persistentLaunchIndex: 0
+
+    anchors.fill: parent
+
+    ListView {
+        id: appsListView
+
+        orientation: ListView.Vertical
+        snapMode: ListView.SnapToItem
+        width: parent.width > parent.height ? parent.height : parent.width
+        height: parent.height
+        anchors.centerIn: parent
+        clip: true
+        cacheBuffer: height * 2
+
+        property int currentPos: 0
+
+        onCurrentPosChanged: {
+            rightIndicator.animate()
+            leftIndicator.animate()
+            topIndicator.animate()
+            bottomIndicator.animate()
+        }
+
+        onCurrentIndexChanged: {
+            // Update the persistentLaunchIndex when user scrolls to a new position
+            if (grid.currentVerticalPos === 1) {  // Only when in the launcher view
+                root.persistentLaunchIndex = currentIndex
+            }
+        }
+
+        onAtYBeginningChanged: {
+            if ((grid.currentHorizontalPos === 0) && (grid.currentVerticalPos === 1)) {
+                forbidTop = !atYBeginning
+                grid.changeAllowedDirections()
+            }
+        }
+
+        Connections {
+            target: grid
+            function onCurrentVerticalPosChanged() {
+                if (grid.currentVerticalPos === 0) {
+                    appsListView.highlightMoveDuration = 0
+                    appsListView.currentIndex = 0
```

Comment:
Why is index tracking even needed? Removing it above lines (https://github.com/MagneFire/asteroid-launcher/commit/f369524a88c9faa86b3fbc0e570d4e9989df6906) also works :D

Reply:
lastLaunchedIndex: Tracks the last app that the user actually launched
persistentLaunchIndex: Tracks the position in the launcher view regardless of app launch

However, this can be simplified by using only lastLaunchedIndex since i did not understand i can use it for both cases when i implemented that.
Fix incoming.

Reply:
I understand what they track, but why is there a reason for it in the first place? If you never reset the `ListView` `currentIndex` there's no need track `lastLaunchedIndex` and `persistentLaunchIndex`.

Let me phrase it this way. When do you want to show the beginning of the view?

Reply:
The first item is only shown as entry point when user willingly left the launcher by scrolling all the way up.
In any other case, the lastLaunchedIndex shall be respected afaiu.

And i only explained to make it clear to myself that the persistentLaunchIndex was redundand.
I refactored to only use lastLaunchedIndex.